Api connection

nano1342 opened the pull request.

nano1342 committed:
made first working route

nano1342 committed:
stuck at CORS error of addSong

nano1342 committed:
fixed CORS problem

nano1342 committed:
completed artists routes

nano1342 committed:
completed all album routes

nano1342 committed:
completed tags routes

Picorims commented on src/models/Tag.ts:
There is already an interface for this.

Picorims commented on src/models/Song.ts:
There is already an interface for this.

Picorims committed:
🦄 refactor(api): give distinct name to API bound models

This is to distinguish them with front and save based models

Picorims committed:
🌈 style(api): flag incorrect usage of any to fix

not fixed right away as there are higher priorities on the front depending of this branch

Picorims committed:
Merge branch 'main' into API-connection
